src/storage/chaindb.md

```diff
@@ -0,0 +1,84 @@
+# The ChainDB format
+
+The format in which the Cardano blockchain is stored in the disk has been historically the ChainDB format used by IOG's `cardano-node`. It consists of 3 separate directories in which different data is stored.
+
+```mermaid
+flowchart RL
+    subgraph volatile;
+      subgraph "current selection";
+        Tip -->|...| Km1["(k-1)-th Block"]
+        Km1 --> K["k-th Block"]
+      end
+
+      O1["Block"] --> O2["Block"]
+      O3["Block"] --> K
+
+      O4["Block"]
+    end
+
+    subgraph immutable;
+     subgraph "Chunk 0";
+        C0[" "]
+        C1[" "] --> C0
+        C2[" "] -->|...| C1
+     end
+     subgraph "Chunk 1";
+        D0[" "]
+        D1[" "] --> D0
+        D2[" "] -->|...| D1
+     end
+
+     D0 --> C2
+     subgraph "Chunk n";
+     Kp1["(k+1)-th Block"] --> Kp2["(k+2)-th Block"]
+     Kp2["(k+2)-th Block"] --> |...| E0[" "]
+     end
+
+     E0 -->|...| D2
+    end
+    K --> Kp1
+    C0 --> Genesis
+    subgraph ledger;
+     S["Persisted snapshot"] --> Kp1
+    end
+```
+
+## `immutable`
+
+Contains the historical chain of blocks.
+
+TODO explain NestedCtxt, Header, Block, the Hard Fork Block.
+
+## `volatile`
+
+Contains blocks that form the current selection of the node (i.e. the best chain it knows about) and other blocks (both connected or disconnected from the selected chain) but whose slot number is greater than the `k`-th block in the current selection (so they might belong to a fork less than `k` blocks deep).
+
+TODO describe the format in which the blocks are stored
+
+## `ledger`
+
+Contains snapshots of the ledger state at some immutable block, ideally the last one but as ledger snapshotting is done periodically it might belong to some older block. The snapshot is named after the slot number of the block at which the snapshot was taken.
```

Comment:
```suggestion
Contains ledger state snapshots at immutable blocks. Ideally, this is the most recent immutable block, but since snapshots are taken periodically (due to their cost), it may be an older block. Snapshots are named after the slot number of the block when they were taken.
```